### DISCUSSION TITLE
Regression or classification task does not respect split type

### DISCUSSION THREAD

**Discussion by jmrichardson** [microsoft/FLAML#1340] [General]
Even though split type is time and eval method is holdout, for regression or classification it will always shuffle the split

```
 self.automl.fit(
            dataframe=df,
            label="target",
            time_col="date",
            # time_budget=self.time_budget,
            time_budget=200,
            task='regression',
            ensemble=True,
            estimator_list=['lgbm', 'extra_tree'],
            eval_method="holdout",
            split_type="time",
            n_splits=1,
            split_ratio=.3,
            retrain_full=True,
            log_file_name='automl_checkpoint.log',
        )
```

in generic_task.py, this block of code will ultimately split the data:

            elif self.is_regression():
                X_train, X_val, y_train, y_val = self._train_test_split(
                    state, X_train_all, y_train_all, split_ratio=split_ratio
                )

**Comment by jmrichardson** [comment]
I think Line 477  in generic_task should be:

`elif split_type == "group":`

instead of:

`if split_type == "group":`

**Comment by jmrichardson** [comment]
No response.  Too many bugs...
